openscvx/plotting/plotting.py

```diff
@@ -1,279 +1,298 @@
-        if params is None:
-            params = problem.settings
+    dim = _get_var_dim(result, state_name, result._states)
+    t_nodes = result.nodes["time"].flatten()
+    has_trajectory = bool(result.trajectory) and state_name in result.trajectory
+    t_full = result.trajectory["time"].flatten() if has_trajectory else None
 
-    if result is None or params is None:
-        raise ValueError("Must provide either (result, params) or problem")
+    fig = go.Figure()
+    fig.update_layout(title_text=state_name, template="plotly_dark")
 
-    # Optional filtering of which states to plot
-    state_filter = set(state_names) if state_names else None
+    for i in range(dim):
+        label = f"{state_name}_{i}" if dim > 1 else state_name
 
-    # Get time values at nodes from the nodes dictionary
-    t_nodes = result.nodes["time"].flatten()
+        if show in ("both", "trajectory") and has_trajectory:
+            data = result.trajectory[state_name]
+            y = data if data.ndim == 1 else data[:, i]
+            fig.add_trace(go.Scatter(x=t_full, y=y, mode="lines", name=f"{label} (traj)"))
 
-    # Check if full propagation trajectory is available
-    has_full_trajectory = result.trajectory and len(result.trajectory) > 0
+        if show in ("both", "nodes") and state_name in result.nodes:
+            data = result.nodes[state_name]
+            y = data if data.ndim == 1 else data[:, i]
+            fig.add_trace(go.Scatter(x=t_nodes, y=y, mode="markers", name=f"{label} (nodes)"))
 
-    # Get time for full trajectory
-    if has_full_trajectory:
-        t_full = result.trajectory["time"].flatten()
+    fig.update_xaxes(title_text="Time (s)")
+    return fig
 
-    # Get all states (both user-defined and augmented)
-    states = result._states if hasattr(result, "_states") and result._states else []
-
-    # Filter out CTCS augmentation states
-    filtered_states = []
-    for state in states:
-        # Check if this is a CTCS augmentation state (names like _ctcs_aug_0, _ctcs_aug_1, etc.)
-        if "ctcs_aug" not in state.name.lower():
-            filtered_states.append(state)
-
-    states = filtered_states
-
-    if state_filter:
-        states = [s for s in states if s.name in state_filter]
-
-    # Expand states into individual components for multi-dimensional states
-    expanded_states = []
-    for state in states:
-        state_slice = state._slice
-        if isinstance(state_slice, slice):
-            slice_start = state_slice.start if state_slice.start is not None else 0
-            slice_stop = state_slice.stop if state_slice.stop is not None else slice_start + 1
-            n_components = slice_stop - slice_start
-        else:
-            slice_start = state_slice
-            n_components = 1
 
-        # Create a separate entry for each component
-        if n_components > 1:
-            for i in range(n_components):
+def plot_states(
```

Comment:
In cases where there are >1D states or controls we should still plot them each induvidually on there own subplot. Keeping them on the same can be tricky in cases where one of the ND dimensions is scaled differently to the others and also they may not all share the same bounds.